src/dialog-editor/components/dialog-editor/dialogEditorComponent.ts

```diff
@@ -0,0 +1,68 @@
+import * as ng from 'angular';
+
+export class DialogEditorController {
+  public modalOptions: any;
+  public elementInfo: any;
+  public treeSelectorShow: boolean = false;
+  public treeSelectorIncludeDomain: boolean;
+
+  public setupModalOptions(type, tab, box, field) {
+    const components = {
+      tab: 'dialog-editor-modal-tab',
+      box: 'dialog-editor-modal-box',
+      field: 'dialog-editor-modal-field'
+    };
+    this.modalOptions = {
+      component: components[type],
+      size: 'lg',
+    };
+    this.elementInfo = { type: type, tabId: tab, boxId: box, fieldId: field };
+  }
+
+  public treeSelectorToggle() {
+    this.treeSelectorShow = ! this.treeSelectorShow;
+  }
+
+  public treeSelectorSelect(node, elementData) {
+    const fqname = node.fqname.split('/');
+    if (this.treeSelectorIncludeDomain === false) {
+      fqname.splice(1, 1);
+    }
+    elementData.resource_action.ae_instance = fqname.pop();
```

Comment:
Please reafactor it to use [rest operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-1.html)
```javascript
elementData.resource_action = {
  ...elementData.resource_action,
  ae_instance: fqname.pop(),
  ae_class: fqname.pop(),
  ae_namespace: fqname.filter(String).join('/')
}
```